include/onnxruntime/core/framework/allocator.h

```diff
@@ -130,29 +130,32 @@ class IAllocator {
      Create a std::unique_ptr that is allocated and freed by the provided IAllocator.
      @param allocator The allocator.
      @param count_or_bytes The exact bytes to allocate if T is void, otherwise the number of elements to allocate.
+     @param is_transient If true, call Reserve() instead of Alloc() to allocate memory.
```

Comment:
`is_transient` seems confusing. Maybe we should think along the lines of the Torch flag. I think they have it as `non_caching` ?

Reply:
Another optioin is to use 'use_reserve' as that aligns with the Reserve() function exposed in IAllocator and doesn't introduce a new concept that'll need to be explained.